### PR TITLE
release new version v1.0.1 of rescale

### PR DESCRIPTION
**Pull request by wywywywy**
Main repo: https://github.com/wywywywy/kubectl-rescale

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

File: plugins/rescale.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  - selector:
+      matchLabels:
+        os: windows
+        arch: arm64
+    uri: https://github.com/wywywywy/kubectl-rescale/releases/download/v1.0.1/kubectl-rescale_v1.0.1_windows_arm64.tar.gz
+    sha256: 4a7c949418d851efbc69e32df21c364bb116891788d016a8afff46efd33e15a3
+    bin: kubectl-rescale.exe
+  shortDescription: Scale a deployment or statefulset to 0 then back up to the original desired replica count.
+  description: |
+    A kubectl plugin to scale a deployment or statefulset to 0 then back up to the original desired replica count.
```
Comment (ahmetb) [inline plugins/rescale.yaml:53]: what does this plugin do that's nontrivial differently than the user calling "kubectl scale" twice?

File: plugins/rescale.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    Usage:
```
[the remark this answers is not in the document]
Comment (wywywywy) [reply plugins/rescale.yaml]: Noted & updated

File: plugins/rescale.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    sha256: d0d2d5a230fff9c7f53697b343e40a2f1dc255a41ec72b531f8b05247c37a9d4
+    bin: kubectl-rescale.exe
+  - selector:
+      matchLabels:
+        os: windows
+        arch: arm64
+    uri: https://github.com/wywywywy/kubectl-rescale/releases/download/v1.0.1/kubectl-rescale_v1.0.1_windows_arm64.tar.gz
+    sha256: 4a7c949418d851efbc69e32df21c364bb116891788d016a8afff46efd33e15a3
+    bin: kubectl-rescale.exe
+  shortDescription: Scale a deployment or statefulset to 0 then back up to the original desired replica count.
```
[the remark this answers is not in the document]
Comment (wywywywy) [reply plugins/rescale.yaml]: Noted & updated

File: plugins/rescale.yaml (hunk 1 above)
Comment (wywywywy) [reply plugins/rescale.yaml]: Essentially this is a shortcut to do so

Comment (ahmetb) [reply plugins/rescale.yaml]: Typically we don't allow one/two-liner shell functions to be distributed here. Ideally plugins should enhance kubectl experience or be companion tools to it. 

As it stands, I'm -1 about accepting this plugin.

Comment (wywywywy) [reply plugins/rescale.yaml]: No problem @ahmetb. I totally understand, and it's been a good learning experience for me.

I'll close this PR now. Thank you very much for your time.

**Comment by ahmetb** [comment]
You can still distribute your plugin via Krew using custom indexes: https://krew.sigs.k8s.io/docs/developer-guide/custom-indexes/ That way users can still install your plugin via Krew, but they'll be downloading the plugin manifest from your repository instead of krew-index.

**Comment by ahmetb** [comment]
Also upon taking a look at your source code, it seems the logic is a bit more sophisticated than `kubectl scale --to=0 && kubectl scale --to=N`. I initially did not understand that from the description. What's the use case for this sort of command pattern?

**Comment by wywywywy** [comment]
We have quite an unusual use case, which so I built this tool to help with this particular scenario.

Basically we have deployments that cannot use rolling restarts, because it has potential to cause problems during the brief period where both the old version and the new version are running.  

I understand that we can achieve this in many ways without a custom tool, but we also have another annoyance - we have something deployed with the same name as a deployment in some namespaces, but as a statefulset in other namespaces. E.g. `deployment/crawler` in `customer1` & `statefulset/crawler` in `customer2`. 

With this tool I can do a restart simply by doing `kubectl rescale crawler -n customer1` without first finding out whether it's a deployment or statefulset.

I understand it's not a common thing, and is very specific to our environment.

**Pull request closed without merging**